pkg/testing/fetcher_artifact.go

```diff
@@ -68,7 +68,9 @@ func (f *artifactFetcher) Fetch(ctx context.Context, operatingSystem string, arc
 	}
 	preVersion := version
 	if uri == "" {
-		version = fmt.Sprintf("%s-SNAPSHOT", version)
+		if !strings.HasSuffix(version, "-SNAPSHOT") {
+			version += "-SNAPSHOT"
+		}
```

Comment:
We actually did force it there. What didn't work was when you passed in a version string that had the suffix `-SNAPSHOT` it would append it again, which resulted in `-SNAPSHOT-SNAPSHOT` which was incorrect.